fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/DimensionRow.java

```diff
@@ -16,6 +17,21 @@
 
     private final String keyValue;
 
+
+    /**
+     * Build a dimension row with a key field value and a map of field values.
+     *
+     * @param key  the keyField for this row
+     * @param dimensionValue  the value for the key
+     */
+    public DimensionRow(@NotNull DimensionField key, String dimensionValue) {
+        super(Collections.singletonMap(key, dimensionValue));
+        if (key == null) {
+            throw new IllegalArgumentException("Missing key " + key);
```

Comment:
`key` will always be `null`. We could change message to `String.format("Missing key for dimension value '%s'", dimensionValue)`